cni/network/network.go

```diff
@@ -425,41 +423,47 @@ func (plugin *NetPlugin) Add(args *cniSkel.CmdArgs) error {
 	}
 
 	if nwCfg.MultiTenancy {
-		cnsclient, er := cnscli.New(nwCfg.CNSUrl, defaultRequestTimeout)
-		if err != nil {
+		cnsClient, er := cnscli.New(nwCfg.CNSUrl, defaultRequestTimeout)
+		if er != nil {
 			return fmt.Errorf("failed to create cns client for multitenancy %w", er)
 		}
-		plugin.multitenancyClient.Init(cnsclient, AzureNetIOShim{})
+
 		plugin.report.Context = "AzureCNIMultitenancy"
+		plugin.multitenancyClient.Init(cnsClient, AzureNetIOShim{})
+		plugin.ipamInvoker = NewCNSInvoker(k8sPodName, k8sNamespace, cnsClient, plugin.multitenancyClient)
+
 		// Temporary if block to determining whether we disable SNAT on host (for multi-tenant scenario only)
-		if enableSnatForDns, nwCfg.EnableSnatOnHost, err = plugin.multitenancyClient.DetermineSnatFeatureOnHost(
+		if enableSnatForDNS, nwCfg.EnableSnatOnHost, err = plugin.multitenancyClient.DetermineSnatFeatureOnHost(
 			snatConfigFileName, nmAgentSupportedApisURL); err != nil {
 			return fmt.Errorf("%w", err)
 		}
 
-		result, cnsNetworkConfig, subnetPrefix, azIpamResult, err = plugin.GetMultiTenancyCNIResult(
-			context.TODO(), enableInfraVnet, nwCfg, k8sPodName, k8sNamespace, args.IfName)
-		if err != nil {
-			log.Printf("GetMultiTenancyCNIResult failed with error %v", err)
-			return fmt.Errorf("GetMultiTenancyCNIResult failed:%w", err)
+		ipamAddResult.ipv4Result, ipamAddResult.ncResponse, ipamAddResult.hostSubnetPrefix, er = plugin.multitenancyClient.GetContainerNetworkConfiguration(
```

Comment:
should not be returning result from GetContainerNetworkConfiguration as we discussed in call

Reply:
why are we changing the return here? (we use it throughout the rest of func and the call didn't explain why)

Reply:
removed and added conversion underneath via convertToCniResult